packages/@lwc/engine-core/src/framework/restrictions.ts

```diff
@@ -263,23 +269,6 @@ function getCustomElementRestrictionsDescriptors(elm: HTMLElement): PropertyDesc
     };
 }
 
-function getComponentRestrictionsDescriptors(): PropertyDescriptorMap {
-    assertNotProd(); // this method should never leak to prod
-    return {
-        tagName: generateAccessorDescriptor({
-            get(this: LightningElement) {
-                throw new Error(
-                    `Usage of property \`tagName\` is disallowed because the component itself does` +
-                        ` not know which tagName will be used to create the element, therefore writing` +
-                        ` code that check for that value is error prone.`
-                );
-            },
-            configurable: true,
-            enumerable: false, // no enumerable properties on component
-        }),
-    };
-}
```

Comment:
This one also gives an `Illegal invocation` error when I try to invoke the original descriptor's `get`, so I don't see the point in keeping it around.